From: Michal Hocko <mhocko@suse.com>
Subject: kabi, fs/seq_file: fallback to vmalloc allocation
Patch-mainline: never, kabi fixup
References: bnc#968687

The kabi checker doesn't like to see mm.h include because some of the types
will get expended and so it looks like they have changed.

KABI: symbol __seq_open_private(vmlinux) changed crc from 0xfdda6269 to 0xc4d22967
KABI: symbol seq_lseek(vmlinux) changed crc from 0xe5e1e6e5 to 0x79fb31b5
KABI: symbol seq_open(vmlinux) changed crc from 0x65e3fc04 to 0xe6da9c4f
KABI: symbol seq_open_private(vmlinux) changed crc from 0xb522205d to 0x5671f0da
KABI: symbol seq_path(vmlinux) changed crc from 0x82c77e59 to 0xe7613fc6
KABI: symbol seq_read(vmlinux) changed crc from 0x7f793823 to 0xe751bc27
KABI: symbol seq_release(vmlinux) changed crc from 0xb732d84f to 0xd3e49a6a
KABI: symbol seq_release_private(vmlinux) changed crc from 0x3839a11d to 0xa4f10017
KABI: symbol single_open(vmlinux) changed crc from 0x8bad7ad5 to 0xaf97348a
KABI: symbol single_open_size(vmlinux) changed crc from 0xf0941d8f to 0xb999d7c0
KABI: symbol single_release(vmlinux) changed crc from 0xde83daab to 0x8d951c38

This is a deficiency of the checker and not a real kABI breakage. Paper over it
and hide the include for the kabi check.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 fs/seq_file.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/seq_file.c
+++ b/fs/seq_file.c
@@ -10,7 +10,9 @@
 #include <linux/seq_file.h>
 #include <linux/vmalloc.h>
 #include <linux/slab.h>
+#ifndef __GENKSYMS__
 #include <linux/mm.h>
+#endif
 
 #include <asm/uaccess.h>
 #include <asm/page.h>
